openlibrary/templates/lib/header_dropdown.html

```diff
@@ -0,0 +1,27 @@
+$def with (props)
```

Comment:
It may make it easier for the person converting this to Vue later if the required and optional props were documented here.

Reply:
this was a great idea. Done.

Reply:
```suggestion
$def with (props)
$# @typedef {Object} DropdownLink
$# @property {string} text of link
$# @property {string|null} href of link for HTTP get request
$# @property {string|null} post when set will be used instead of href and should use HTTP POST request.
$#
$# @param {Object} props template properties
$# @param {string} props.name unique identifying name for dropdown and distinguishing it from other dropdowns
$# @param {string} props.label for menu, also used as alt text for dropdown icon when props.image is set
$# @param {string|null} props.image URL for image (optional)
$# @param {DropdownLink[]} props.links
```

It looks like comments have to appear after the initial `$def with` statement in Templetor, else an error is thrown.